src/clang_parser.cpp

```diff
@@ -230,8 +230,25 @@ void ClangParser::parse(ast::Program *program, BPFtrace &bpftrace, std::vector<s
       &translation_unit);
   if (error)
   {
-    std::cerr << "Clang error while parsing C definitions: " << error << std::endl;
-    std::cerr << "Input (" << input.size() << "): " << input << std::endl;
+    if (bt_debug == DebugLevel::kFullDebug) {
+      std::cerr << "Clang error while parsing C definitions: " << error << std::endl;
+      std::cerr << "Input (" << input.size() << "): " << input << std::endl;
+    }
+    clang_disposeTranslationUnit(translation_unit);
```

Comment:
We shouldn't have multiple exit points where we have to include these clean up lines - they are the kind of thing that could get forgotten in the future

Reply:
I tried to come up with a solution to that with minimal code changes but had no success. Let me give it another shot.

Reply:
Done, the second commit addresses this issue. Let me know if that approach works for you. If it does, I'll squash both commits.

Reply:
Thanks, I'll read it over tomorrow